test/functional/rpc_netinfo.py

```diff
@@ -286,13 +453,108 @@ def test_deprecation(self):
+        # to auto-fill the addr portion from coreP2PAddrs[0]
+        proupservtx_hash = self.node_evo.update_mn(self, True,
+                                                   [f"127.0.0.1:{self.node_evo.mn.nodePort}", f"[::1]:{self.node_evo.mn.nodePort}"],
+                                                   DEFAULT_PORT_PLATFORM_P2P + 12,
+                                                   [f"127.0.0.3:{DEFAULT_PORT_PLATFORM_HTTP}", f"[::3]:{DEFAULT_PORT_PLATFORM_HTTP}"])
+        proupservtx_rpc = self.node_evo.node.getrawtransaction(proupservtx_hash, True)
+        assert_equal(proupservtx_rpc['proUpServTx']['addresses']['platform_p2p'][0], f"127.0.0.1:{DEFAULT_PORT_PLATFORM_P2P + 12}")
+
+        # Since all fields are stored in the netInfo, even though we updated only platformP2PAddrs, the dummy address should never be visible
+        # and *all* the addresses fields will be visible as they're now in one data structure
+        proupservtx_height = proupservtx_rpc['height']
+        protx_listdiff_rpc = self.node_evo.node.protx('listdiff', proupservtx_height - 1, proupservtx_height)
+        protx_listdiff_rpc = self.extract_from_listdiff(protx_listdiff_rpc, proregtx_hash)
+        assert_equal(protx_listdiff_rpc['addresses'], {
+            'core_p2p': [f"127.0.0.1:{self.node_evo.mn.nodePort}", f"[::1]:{self.node_evo.mn.nodePort}"],
+            'platform_p2p': [f"127.0.0.1:{DEFAULT_PORT_PLATFORM_P2P + 12}"],
+            'platform_https': [f"127.0.0.3:{DEFAULT_PORT_PLATFORM_HTTP}", f"[::3]:{DEFAULT_PORT_PLATFORM_HTTP}"]
+        })
+
+        # Check platform{HTTP,P2P}Port shims work as expected
+        assert_equal(proregtx_rpc['proRegTx']['platformP2PPort'], DEFAULT_PORT_PLATFORM_P2P)
+        assert_equal(proregtx_rpc['proRegTx']['platformHTTPPort'], DEFAULT_PORT_PLATFORM_HTTP)
+        assert_equal(proupservtx_rpc['proUpServTx']['platformP2PPort'], DEFAULT_PORT_PLATFORM_P2P + 12)
+        assert_equal(proupservtx_rpc['proUpServTx']['platformHTTPPort'], DEFAULT_PORT_PLATFORM_HTTP)
+        # platform{HTTP,P2P}Port *won't* be populated by listdiff as that *field* is now unused and it's dangerous to report "updates" to disused fields
+        assert "platformP2PPort" not in protx_listdiff_rpc.keys()
+        assert "platformHTTPPort" not in protx_listdiff_rpc.keys()
+
+        # Restart the client to see if (de)ser works as intended (CDeterministicMNStateDiff is a special case and we just made an update)
+        self.node_evo.set_active_state(self, False)
+        self.reconnect_nodes()
+
+        # Check that 'service' correctly reports as coreP2PAddrs[0]
+        proregtx_rpc = self.node_simple.getrawtransaction(proregtx_hash, True)
+        proupservtx_rpc = self.node_simple.getrawtransaction(proupservtx_hash, True)
+        assert_equal(proregtx_rpc['proRegTx']['service'], f"127.0.0.1:{self.node_evo.mn.nodePort}")
+        assert_equal(proupservtx_rpc['proUpServTx']['service'], f"127.0.0.1:{self.node_evo.mn.nodePort}")
```

Comment:
(out-of-scope, refactoring suggestion)
`rpc_netinfo.py` is really long now! Is there any way to make it more compact? maybe de-duplicate code somehow?